marklogic-data-hub/src/main/java/com/marklogic/hub/collector/impl/CollectorImpl.java

```diff
@@ -96,147 +64,29 @@ public CollectorImpl(HubClient hubClient, String sourceDatabase) {
-                    @Override
-                    public boolean verify(String paramString, SSLSession paramSSLSession) {
-                        return true;
-                    }
-
-                    @Override
-                    public void verify(String host, SSLSocket ssl) throws IOException {
-                    }
-
-                    @Override
-                    public void verify(String host, X509Certificate cert) throws SSLException {
-                    }
-
-                    @Override
-                    public void verify(String host, String[] cns, String[] subjectAlts) throws SSLException {
-                    }
-                };
-
-            } else if (verifier == SSLHostnameVerifier.COMMON) {
-                hostnameVerifier = null;
-            } else if (verifier == SSLHostnameVerifier.STRICT) {
-                hostnameVerifier = null;
-            } else if (verifier != null) {
-                hostnameVerifier = new HostnameVerifierAdapter(verifier);
-            }
-            httpClientBuilder.setHostnameVerifier(hostnameVerifier);
-        }
-
-        HttpClient client = httpClientBuilder.build();
-
-        RestTemplate rt = new RestTemplate(new HttpComponentsClientHttpRequestFactory(client));
-        rt.setErrorHandler(new MgmtResponseErrorHandler());
-        return rt;
-    }
-
-    static private class HostnameVerifierAdapter implements X509HostnameVerifier {
-        private SSLHostnameVerifier verifier;
-
-        protected HostnameVerifierAdapter(SSLHostnameVerifier verifier) {
-            this.verifier = verifier;
-        }
-
-        public void verify(String hostname, X509Certificate cert) throws SSLException {
-            ArrayList<String> cnArray = new ArrayList<>();
-            try {
-                LdapName ldapDN = new LdapName(cert.getSubjectX500Principal().getName());
-                for (Rdn rdn : ldapDN.getRdns()) {
-                    Object value = rdn.getValue();
-                    if ("CN".equalsIgnoreCase(rdn.getType()) && value instanceof String) {
-                        cnArray.add((String) value);
-                    }
-                }
-                int type_dnsName = 2;
-                int type_ipAddress = 7;
-                ArrayList<String> subjectAltArray = new ArrayList<>();
-                Collection<List<?>> alts = cert.getSubjectAlternativeNames();
-                if (alts != null) {
-                    for (List<?> alt : alts) {
-                        if (alt != null && alt.size() == 2 && alt.get(1) instanceof String) {
-                            Integer type = (Integer) alt.get(0);
-                            if (type == type_dnsName || type == type_ipAddress) {
-                                subjectAltArray.add((String) alt.get(1));
-                            }
-                        }
-                    }
+            } finally {
+                if (bufferedReader != null) {
```

Comment:
We don't need the check since `IOUtils.closeQuietly` will check and close.

Reply:
Antoher option would be to use a try-with-resource statement, since the bufferedReader isn't required outside of the try block. https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
This PR also requires a rebase